jup_interface/src/pool_sol/deposit_withdraw_sol.rs

```diff
@@ -0,0 +1,176 @@
+use anyhow::{anyhow, Result};
+use jupiter_amm_interface::{
+    AccountMap, Amm, AmmContext, KeyedAccount, Quote, QuoteParams, Swap, SwapAndAccountMetas,
+    SwapParams,
+};
+use solana_sdk::{instruction::AccountMeta, pubkey::Pubkey, system_program};
+use spl_token::native_mint;
+use stakedex_interface::{
+    StakeWrappedSolKeys, WithdrawWrappedSolKeys, STAKE_WRAPPED_SOL_IX_ACCOUNTS_LEN,
+    WITHDRAW_WRAPPED_SOL_IX_ACCOUNTS_LEN,
+};
+use stakedex_sdk_common::{
+    find_deposit_stake_amm_key, find_fee_token_acc, stakedex_program, wsol_bridge_in, DepositSol,
+    InitFromKeyedAccount, WithdrawSol, TEMPORARY_JUP_AMM_LABEL,
+};
+
+use crate::jupiter_stakedex_interface::STAKEDEX_ACCOUNT_META;
+
+// newtype pattern in order to impl external trait (Amm) on external generic (WithdrawSol)
+#[derive(Clone)]
+pub struct DepositWithdrawSolWrapper<T: WithdrawSol + Clone + Send + Sync + 'static>(pub T);
```

Comment:
```suggestion
#[derive(Clone)]
pub struct DepositWithdrawSolWrapper<T>(pub T);
```

trait bounds on implementation should suffice, its missing a `+ DepositSol` on the bound otherwise. Will probably refactor to remove the bounds for all the other wrappers at some point in the future